zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -164,10 +183,21 @@ pub(crate) fn select_spendable_sapling_notes(
     anchor_height: BlockHeight,
     exclude: &[ReceivedNoteId],
 ) -> Result<Vec<ReceivedSaplingNote<ReceivedNoteId>>, SqliteClientError> {
+    let mut stmt_unscanned_tip = conn.prepare_cached(
+        "SELECT 1 FROM v_sapling_shard_unscanned_ranges unscanned
+         WHERE unscanned.shard_index IN (SELECT MAX(shard_index) FROM sapling_tree_shards)
+         AND unscanned.subtree_start_height <= :anchor_height",
+    )?;
+    let mut unscanned =
+        stmt_unscanned_tip.query(named_params![":anchor_height": &u32::from(anchor_height),])?;
+    if unscanned.next()?.is_some() {
+        // if the tip shard has unscanned ranges below the anchor height, none of our notes can be
+        // spent
+        return Ok(vec![]);
+    }
+
     // The goal of this SQL statement is to select the oldest notes until the required
-    // value has been reached, and then fetch the witnesses at the desired height for the
-    // selected notes. This is achieved in several steps:
-    //
+    // value has been reached.
```

Comment:
```suggestion
    // value has been reached.
    //
```

Reply:
"up to" on lines 204 and 206 mean "up to and including", correct? Please make that explicit.

Reply:
Point 3 does not take into account that a note can have zero value. Thus, taking notes with cumulative value less than the required value and then one extra note, does not necessarily satisfy the required value. It is necessary to take notes with cumulative value **less than or equal to** the required value, and then one extra note (which necessarily then has a non-zero value) if needed.

Reply:
Also, taking a note may require an increase in the fee. I'm not sure whether that is handled at a higher level.

Reply:
On further inspection, I don't think this is a bug:

https://github.com/zcash/librustzcash/pull/894/files#diff-9cea07a2b90189d927a2737a964494dfea6ab1bc5a51342e481210b93586716eR218-R219

creates a running sum over the rows, and then 

https://github.com/zcash/librustzcash/pull/894/files#diff-9cea07a2b90189d927a2737a964494dfea6ab1bc5a51342e481210b93586716eR237-R238

ensures that a row is returned (if it exists) such that the running sum is greater than or equal to the required amount.